main/src/ca/uwaterloo/flix/language/phase/Typer.scala

```diff
@@ -323,6 +321,19 @@ object Typer {
       }
   }
 
+  /**
+    * Performs processing on the type variable names to attempt to preserve user-given names.
```

Comment:
Computes an equ-most general substitution with the given `tparams` as rigid variables.